tests/_test_msui/test_multiple_flightpath_dockwidget.py

```diff
@@ -25,33 +25,244 @@
     limitations under the License.
 """
 import pytest
-from PyQt5 import QtTest
+from PyQt5 import QtWidgets, QtCore, QtGui, QtTest
+from unittest import mock
+
 from mslib.msui import msui
-from mslib.msui.multiple_flightpath_dockwidget import MultipleFlightpathControlWidget
-from mslib.msui import flighttrack as ft
 import mslib.msui.topview as tv
 
 
-class Test_MultipleFlightpathControlWidget:
-    @pytest.fixture(autouse=True)
-    def setup(self, qtbot):
-        self.window = msui.MSUIMainWindow()
-        self.window.create_new_flight_track()
-
-        self.window.actionNewFlightTrack.trigger()
-        self.window.actionTopView.trigger()
-        initial_waypoints = [ft.Waypoint(40., 25., 0), ft.Waypoint(60., -10., 0), ft.Waypoint(40., 10, 0)]
-        self.waypoints_model = ft.WaypointsTableModel("myname")
-        self.waypoints_model.insertRows(
-            0, rows=len(initial_waypoints), waypoints=initial_waypoints)
-
-        self.widget = tv.MSUITopViewWindow(model=self.waypoints_model, mainwindow=self.window)
-        self.window.show()
-        QtTest.QTest.qWaitForWindowExposed(self.window)
-        yield
-        self.window.hide()
-
-    def test_initialization(self):
-        widget = MultipleFlightpathControlWidget(parent=self.widget,
-                                                 listFlightTracks=self.window.listFlightTracks)
-        assert widget.color == (0, 0, 1, 1)
+@pytest.fixture
+def main_window(qtbot):
+    """
+    Set-up for the docking widget
+    """
+    # Start a MSUI window
+    window = msui.MSUIMainWindow()
+    window.show()
+    qtbot.wait_exposed(window)
+
+    # Create two flight tracks
+    window.actionNewFlightTrack.trigger()
+    window.actionNewFlightTrack.trigger()
+
+    # Open a Top View window
+    window.actionTopView.trigger()
+    topview_window = window.listViews.currentItem().window
+
+    # Switch to the Multiple Flightpath Widget
+    topview_window.cbTools.setCurrentIndex(6)
+
+    # Get a reference to the created MultipleFlightpathControlWidget
+    multiple_flightpath_widget = topview_window.docks[tv.MULTIPLEFLIGHTPATH].widget()
+
+    yield window, multiple_flightpath_widget
+
+    window.hide()
+
+
+def test_initialization(main_window):
+    """
+    test for conforming docking widget has initialized
+    """
+    _, multiple_flightpath_widget = main_window
+
+    # Ensure the MultipleFlightpathControlWidget is correctly initialized
+    assert multiple_flightpath_widget is not None
+    assert multiple_flightpath_widget.color == (0, 0, 1, 1)
+
+
+def test_setColour(main_window):
+    """
+    test for the filghttrack colour
+    """
+    _, multiple_flightpath_widget = main_window
+    color_button = multiple_flightpath_widget.pushButton_color
+
+    # Mock the exec_ method and color_selected signal of the CustomColorDialog
+    with mock.patch("mslib.utils.colordialog.CustomColorDialog.exec_",
```

Comment:
Why not do the mock-patching as fixture? I just double checked the code and "your" syntax is commonly used by other peoples tests. See
 _tests/_test_msui/test_satellite_dockwidget.py
as example.
I think this is more elegant (similar to your rewriting of setup as fixture; nice!)

Reply:
I am not quite sure if I understand you correctly, are you suggesting to do the mock as a decorator on the test function? I.e. with
```python
@mock.patch(...)
def test_...():
```
, as is used in some other places?

If yes, then I'd argue against that. The mock should only be in effect for the action that needs something to be mocked. This is easy to achieve using a contextmanager like here, while using it as a decorator will apply the mock broadly for the entirety of the test function.

I think it is most readable to use it as a contextmanager and explain right there in a comment why the mock is necessary and no alternative was usable.